tests/test_update.py

```diff
@@ -98,14 +105,15 @@ def test_update_from_same_commit(tmp_path: Path, package_name, branch):
     assert_repo_is_not_dirty(sg_path)
 
     sg_repo = git.Repo(path=sg_path)
-    first_round_commits = list(sg_repo.iter_commits("sg-start..HEAD"))
+    upstream_ref = START_TAG_TEMPLATE.format(branch=branch)
+    first_round_commits = list(sg_repo.iter_commits(f"{upstream_ref}..HEAD"))
 
     run_dist2src(
         ["-vvv", "convert", f"{dist_git_path}:{branch}", f"{sg_path}:{branch}"]
     )
     assert_repo_is_not_dirty(sg_path)
 
-    second_round_commits = list(sg_repo.iter_commits("sg-start..HEAD"))
+    second_round_commits = list(sg_repo.iter_commits(f"{upstream_ref}..HEAD"))
```

Comment:
I was thinking if this is a good-enough test case for this feature or we need to specifically check that the tag is applied on an expected commit.

Reply:
I would say yes: yesterday this helped me identify an issue with the original implementation of this PR.